web/packages/teleterm/src/mainProcess/agentDownloader/agentDownloader.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { pipeline } from 'node:stream/promises';
+import { createReadStream } from 'node:fs';
+import { join } from 'node:path';
+import { createUnzip } from 'node:zlib';
+import { exec } from 'node:child_process';
+import { promisify } from 'node:util';
+
+import { extract } from 'tar-fs';
+
+import { compareSemVers } from 'shared/utils/semVer';
+
+import Logger from 'teleterm/logger';
+
+import { RuntimeSettings } from '../types';
+
+import type { IFileDownloader } from './fileDownloader';
+
+const TELEPORT_CDN_ADDRESS = 'https://cdn.teleport.dev';
+const logger = new Logger('agentDownloader');
+
+interface AgentBinaryParams {
```

Comment:
FWIW, this comment is going to sound like a very academical nit, but when defining an object like this, I think it's good to think of it as a first-class citizen, like we do with document types for example.

By naming it `SomethingParams`, it makes it seem like it exists only to accommodate for `createAgentBinaryName` arguments and serves as replacement for the lack of keyword arguments in JavaScript.

If someone ever needs to extend `createAgentBinaryName` with an optional argument, I bet they're going to just put more stuff into `AgentBinaryParams`. But IMHO the correct way would be to treat `AgentBinaryParams` as this separate thing that describes the agent and add another object to the signature of `createAgentBinaryName`.

As such, if you want to keep this object around and define it as its own thing, I think simply dropping `Params` from the type name would make a better job at explaining that it's a separate object rather than just a wrapper for `createAgentBinaryName` arguments.